### PR TITLE
docs: update `liquidstaking` spec docs

### PR DESCRIPTION
**Pull request by kogisin**
## Description

This PR updates spec documents for `liquidstaking` module.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Appropriate labels applied
- [ ] Targeted PR against correct branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://go.dev/blog/godoc).
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by dongsam — APPROVED** [review]

File: x/liquidstaking/spec/01_concepts.md [hunk 1]
```diff
@@ -4,8 +4,38 @@
...
+- the (re)delegator already has another immature redelegation in progress with a destination to a validator (let's call it Validator X)
+- and, the (re)delegator is attempting to create a new redelegation where the source validator for this new redelegation is Validator X.
+
+## Restake
+
+The module restakes amount to all active liquid validators that corresponds to their weight when an accumulated reward is over `RewardTrigger` value. 
+
+## Unbonding Period
+
+Liquid stakers who unbond their delegation must wait for the duration of the `UnStakingTime`. It is a chain-specific parameter. During the unbonding period, they are still exposed to being slashed for any liquid validator’s misbehavior.
```
Comment (hallazzang) [inline x/liquidstaking/spec/01_concepts.md:37]: Not sure, but:

```suggestion
Liquid stakers who unbond their delegation must wait for the duration of the `UnbondingTime`. It is a chain-specific parameter. During the unbonding period, they are still exposed to being slashed for any liquid validator’s misbehavior.
```

**Comment by CoinedLiberty** [comment]
suggestion

05_begin_block.md

## Rebalancing (Auto-Redelegation)

Due to the events like slashing, tombstoning, become Inactive...

-> Due to the events like slashing, tombstoning, becoming inactive...

**Review by hallazzang — CHANGES_REQUESTED** [review]
It must have been hard work. Great work done! I have a minor change request:

File: x/liquidstaking/spec/08_params.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+| RebalancingTrigger | string (sdk.Dec) | "0.001000000000000000" |
+| RewardTrigger      | string (sdk.Dec) | "0.001000000000000000" |
+
+## RebalancingTrigger
+
+It is the maximum difference and required rate that triggers asset rebalancing (redelegation) for all liquid validators.
+
+## RewardTrigger
+
+It is the rate that triggers to withdraw rewards and re-stake amounts to active validators. Specifically, if the sum of balances including the withdrawn rewards, crumb, and the upcoming rewards of `LiquidStakingProxyAcc` exceeds the rate of `RewardsTrigger` of the total `DelShares`, the rewards are automatically withdrawn and re-stake according to each validator's weight.
```
Comment (hallazzang) [inline x/liquidstaking/spec/08_params.md:52]:
```suggestion
It is the rate that triggers to withdraw rewards and re-stake amounts to active validators. Specifically, if the sum of balances including the withdrawn rewards, crumb, and the upcoming rewards of `LiquidStakingProxyAcc` exceeds the rate of `RewardTrigger` of the total `DelShares`, the rewards are automatically withdrawn and re-stake according to each validator's weight.
```

**Comment by kogisin** [comment]
@CoinedLiberty I actually didn't update `05_begin_block.md` much, but let me update that with other suggestions. Thanks!

File: x/liquidstaking/spec/01_concepts.md (hunk 1 above)
Comment (kogisin) [reply x/liquidstaking/spec/01_concepts.md:37]: I don't even know where `UnStakingTime` comes from. Thanks! Let me change that.

**Review by hallazzang — APPROVED** [review]
LGTM!

**Pull request merged**